strawberry/union.py

```diff
@@ -252,9 +245,23 @@ def union(
     ... class A: ...
     >>> @strawberry.type
     ... class B: ...
-    >>> strawberry.union("Name", (A, Optional[B]))
+    >>> Annotated[A | B, strawberry.union("Name")]
```

Comment:
Isn't `A | B | None` the equivalent type hint? Or is there a semantic gap between old and new syntax? Since it should be evaluated as `A | (B | None)`